libslide/src/bignum/utils.rs

```diff
@@ -10,6 +13,34 @@ pub fn recast_vec<T, U>(v: Vec<T>) -> Vec<U> {
     unsafe { Vec::from_raw_parts(p as *mut U, len, cap) }
 }
 
+// note we cannot use the above recast since we have a user defined type with
+// an undefined way of storing its memory
+// prereqs: tryfrom is implemented for type coercion and default and clone are implemented
+pub fn recast_user_vec<
+    T: std::clone::Clone,
+    U: std::default::Default + std::clone::Clone + std::convert::TryFrom<T>,
+>(
+    v: Vec<T>,
+) -> Result<Vec<U>, &'static str> {
+    let mut res: Vec<U> = vec![U::default(); v.len()];
+    for i in 0..v.len() {
+        let val = match U::try_from(v[i].clone()) {
+            Ok(val) => val,
+            Err(_err) => return Err(ERR_MSG),
+        };
+        res[i] = val;
+    }
+    Ok(res)
+}
+
+pub fn round_complex_vec(v: Vec<Complex>) -> Vec<Complex> {
+    let mut res = v.clone();
+    for i in 0..v.len() {
+        res[i] = v[i].round()
+    }
+    res
+}
```

Comment:
You don't have to clone since you took ownership of the vec, just declare v as mut. Anyway, I think it would be better to call into_iter and map wherever this is needed rather than having this util